hashes/enigma_machine.py

```diff
@@ -40,7 +48,22 @@ def engine(input_character):
     rotator()
 
 
-if __name__ == "__main__":
+def encode_or_decode(message, token):
```

Comment:
Isn't this just the body of the driver code? Why is it not used in the driver code?

Also, please add type hints